ambuda/views/auth.py

```diff
@@ -5,24 +5,76 @@
 https://www.uxmatters.com/mt/archives/2018/09/signon-signoff-and-registration.php
 """
 
+import uuid
+from datetime import datetime
+from typing import Optional
+
+from dateutil.relativedelta import relativedelta
 from flask import Blueprint, flash, render_template, redirect, url_for
 from flask_login import current_user, login_user, logout_user, login_required
 from flask_wtf import FlaskForm, RecaptchaField
-from werkzeug.security import generate_password_hash
-from wtforms import StringField, PasswordField
+from wtforms import HiddenField, StringField, PasswordField, EmailField
 from wtforms import validators as val
 
 import ambuda.queries as q
 from ambuda import database as db
+from ambuda import mail
 
 
 bp = Blueprint("auth", __name__)
 
 
+MAX_TOKEN_LIFESPAN_IN_HOURS = 3
 # FIXME: redirect to site.index once user accounts are more useful.
 POST_AUTH_ROUTE = "proofing.index"
 
 
+def _create_reset_token(user_id) -> str:
+    raw_token = str(uuid.uuid4())
```

Comment:
We should use `secrets.token_urlsafe` instead, because I believe not all implementations of `uuid.uuid4()` give you random results / uuids are sometimes predictable.